### PR TITLE
scripts: generate 3rd party reporters and services

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Download Readme.MD of 3rd party services and reporters. #3790 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] No tests added but changes were tested on localy running site
- [x] I have added necessary documentation (if appropriate)

## Further comments

- added JSONs with services and reporters definitions
- decoupled generateDocs.js file

### CODE REVIEW

**Comment by christian-bromann** [comment]
Thanks for tackling this. I've seen two things that we might want and can fix:

1) format the images so that they point to the raw image to avoid 404:
Screen Shot 2019-05-26 at 10 12 19 AM

2) degrade headlines at least by 1 to avoid big headlines like:
Screen Shot 2019-05-26 at 10 12 26 AM

You can work on the website by doing:
```sh
$ npm run docs:setup
$ npm run docs:generate
$ npm run docs:build
$ cd website/build/webdriver.io
$ http-server -p 8080
```

**Comment by mgrybyk** [comment]
image

image

File: website/siteConfig.js [hunk 1]
```diff
@@ -56,7 +56,10 @@ module.exports = {
     twitter: true,
     twitterImage: 'img/webdriverio.png',
     twitterUsername: 'webdriverio',
-    scrollToTop: true
+    scrollToTop: true,
+
+    /* Show last update time */
+    enableUpdateTime: true
```
Comment (mgrybyk) [inline website/siteConfig.js]: @WillBrock @christian-bromann this one doesn't work for me :(

Comment (WillBrock) [reply website/siteConfig.js]: @mgrybyk hmm, I was able to get the updated date to display by setting `enabeUpdateTime: true` and then re-generating the docs.

wdio-updated-at

Comment (mgrybyk) [reply website/siteConfig.js]: If it works for you then it should work on site as well. Thanks again

File: scripts/docs-generation/3rdPartyDocs.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+/**
+ * Generate docs for 3rd party reporters and services
+ * @param {object} sidebars website/sidebars
+ */
+exports.generate3rdPartyDocs = async (sidebars) => {
+    for (const [, [namePlural, nameSingular]] of Object.entries(plugins)) {
+        for (const { packageName, title, githubUrl } of packages3rdParty) {
+            const readme = await downloadReadme(githubUrl)
+            const id = `3rd-party-${packageName}`.replace(/@/g, '').replace(/\//g, '-')
```
Comment (christian-bromann) [inline scripts/docs-generation/3rdPartyDocs.js]:
```suggestion
            const id = `${packageName}`.replace(/@/g, '').replace(/\//g, '-')
```

I don't think we need to specify this. Maybe better to attach a paragraph at the end or beginning saying that this is a 3rd party script and issues with it should be posted in the project repo of the 3rd party

File: scripts/docs-generation/3rdPartyDocs.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            fs.writeFileSync(path.join(__dirname, '..', '..', 'docs', `_${id}.md`), doc, { encoding: 'utf-8' })
+
+            if (!sidebars.docs[namePlural]) {
+                sidebars.docs[namePlural] = []
+            }
+
+            // eslint-disable-next-line no-console
+            console.log(`Generated docs for ${packageName}`)
+
+            sidebars.docs[namePlural].push(id)
```
Comment (christian-bromann) [inline scripts/docs-generation/3rdPartyDocs.js]: Currently reporters and services are pushed into both buckets:

```json
"Reporter": [
      "allure-reporter",
      "concise-reporter",
      "dot-reporter",
      "junit-reporter",
      "spec-reporter",
      "sumologic-reporter",
      "3rd-party-wdio-reportportal-reporter",
      "3rd-party-wdio-video-reporter",
      "3rd-party-rpii-wdio-html-reporter",
      "3rd-party-wdio-json-reporter",
      "3rd-party-wdio-mochawesome-reporter",
      "3rd-party-wdio-timeline-reporter",
      "3rd-party-wdio-chromedriver-service",
      "3rd-party-wdio-intercept-service",
      "3rd-party-wdio-zafira-listener-service",
      "3rd-party-wdio-reportportal-service",
      "3rd-party-wdio-docker-service"
    ],
    "Services": [
      "appium-service",
      "applitools-service",
      "browserstack-service",
      "devtools-service",
      "firefox-profile-service",
      "sauce-service",
      "selenium-standalone-service",
      "testingbot-service",
      "3rd-party-wdio-reportportal-reporter",
      "3rd-party-wdio-video-reporter",
      "3rd-party-rpii-wdio-html-reporter",
      "3rd-party-wdio-json-reporter",
      "3rd-party-wdio-mochawesome-reporter",
      "3rd-party-wdio-timeline-reporter",
      "3rd-party-wdio-chromedriver-service",
      "3rd-party-wdio-intercept-service",
      "3rd-party-wdio-zafira-listener-service",
      "3rd-party-wdio-reportportal-service",
      "3rd-party-wdio-docker-service"
    ]
```
which shows everything in the service section:
Screen Shot 2019-05-27 at 9 55 34 AM

**Comment by mgrybyk** [comment]
@christian-bromann 
**3rd party info**
image

Don't really like the style, maybe it should be changed to something different?
image

**fixed sidebar**
image

**Comment by mgrybyk** [comment]
Should I commit `en.json` file? https://github.com/webdriverio/webdriverio/pull/4020/files#diff-e325cd5cbe56ac91a36c6cab5c1ece42

**Comment by christian-bromann** [comment]
> Should I commit `en.json` file?

Doesn't matter that much as we don't do translation yet but it is autogenerated anyway.

**Pull request merged**